Comp2022/src/main/cpp/subsystems/Climber.cpp

```diff
@@ -394,13 +398,13 @@ void Climber::MoveToCalibrate(void)
 void Climber::Calibrate()
 {
     if (m_talonValidCL14)
-    {
         m_motorCL14.SetSelectedSensorPosition(0, 0, kCANTimeout);
-        m_targetInches = 0;
-        m_curInches = 0;
-    }
+
     if (m_talonValidCL15)
         m_motorCL15.SetSelectedSensorPosition(0, 0, kCANTimeout);
+

```

Comment:
Wait, by leaving all these statements out of the talonValid if statements, does that mean we could consider the robot calibrated even if the motors aren't responding? Also, just to clarify, is the general rule that we should only wrap direct calls to the motors in talonValid checks?

Reply:
Yes, that is correct. This would let someone use the benchtop board with either CL14 or CL15 set and run the climber with Motion Magic. The talon valid checks had better be caught immediately at power up in the health tab, or we've got problems that are pretty much unfixable later for any subsystem. The subsystem will be in an unknown state if the valid check fails. Also yes to the general statement that _typically_ only writes to the CAN devices should really be in the talonValid checks. I'm sure there are exceptions.

Reply:
The other way to look at is:  The talonValid checks on reads/writes to CAN devices is there ONLY to allow the software to work even when a motor is not responding (on the benchtop board or by choice in a partially assembled robot). But that requires us to verify they are working at boot up or we will not get any timeout messages from the CAN library.

Reply:
Yeah, that makes sense. We want the code to be able to compile and have as much functionality as possible with any number of motors missing.